File: external/HDF-EOS/hdfeos/samples/subsetpoint.f
c
c 


	program subsetpoint

	integer            status, ptextreg, ptreginfo, ptdetach
	integer            ptextper, ptperinfo, ptclose
	
	integer*4          ptfid, ptid, ptopen, ptattach
	integer*4          size, ptdefboxreg, ptdeftmeper
	integer*4          regionid, periodid

	real*8             cornerlon(2), cornerlat(2)
	real*8             datbuf(128), starttime, stoptime
	
	integer DFACC_READ
	parameter (DFACC_READ=1)

c    
c     Open the HDF point file, "PointFile.hdf"
c 

	ptfid = ptopen("PointFile.hdf", DFACC_READ)

	if (ptfid .NE. -1) then

	   ptid = ptattach(ptfid, "FloatBuoy Point")

	   if (ptid .NE. -1) then
	      cornerlon(1) = -145.
	      cornerlat(1) = -15.
	      cornerlon(2) = -135.
	      cornerlat(2) = 8.



	      regionid = ptdefboxreg(ptid, cornerlon, cornerlat)

	      level = 1
	      status = ptreginfo(ptid, regionid, level, 
     1				 "Longitude,Latitude", size)

	      status = ptextreg(ptid, regionid, level, 
     1	                        "Longitude,Latitude", datbuf)

	      do 100 i=1,size/16
		 write(*,*) i, datbuf(2*i-1),datbuf(2*i)
 100	      continue
	      
c             Time Subsetting	      

	      starttime = 35208757.6d0
	      stoptime = 35984639.2d0
	    
	      periodid = ptdeftmeper(ptid, starttime, stoptime)

	      level = 1
	      status = ptperinfo(ptid, periodid, level, "Time", size)

	      status = ptextper(ptid, periodid, level, "Time", datbuf)
	      
	      do 200 i=1,size/8
		 write(*,*) i, datbuf(i)
 200	      continue
	      
	   endif
	
	   status = ptdetach(ptid)
	   status = ptclose(ptfid)

	endif
	stop
	end

File: external/HDF-EOS/hdfeos/samples/SubsetPoint.c
#include "hdf.h"


main()
{

    intn            i, status;
    
    int32           ptfid, PTid, regionID, periodID, size;
    int32           level;
    
    float64         cornerlon[2], cornerlat[2], starttime, stoptime;
    float64        *datbuf;
    


    /*
     * Open the HDF point file, "PointFile.hdf".
     */

    ptfid = PTopen("PointFile.hdf", DFACC_RDWR);


    if (ptfid != -1)
    {

	PTid = PTattach(ptfid, "FloatBuoy Point");

	if (PTid != -1)
	{
	    cornerlon[0] = -145.;
	    cornerlat[0] = -15.;
	    cornerlon[1] = -135;
	    cornerlat[1] = -8.;

	    regionID = PTdefboxregion(PTid, cornerlon, cornerlat);

	    level = 1;
	    status = PTregioninfo(PTid, regionID, level, 
				  "Longitude,Latitude", &size);
	    datbuf = (float64 *) malloc(size);

	    status = PTextractregion(PTid, regionID, level, 
				  "Longitude,Latitude", (char *) datbuf);

	    for (i=0; i<size/16; i++)
	    {
		printf("%d %lf %lf\n", i, datbuf[2*i], datbuf[2*i+1]);
	    }
	    
	    free(datbuf);


	    starttime = 35208757.6;
	    stoptime = 35984639.2;
	    
	    periodID = PTdeftimeperiod(PTid, starttime, stoptime);

	    level = 1;
	    status = PTperiodinfo(PTid, periodID, level, 
				  "Time", &size);
	    datbuf = (float64 *) malloc(size);

	    status = PTextractperiod(PTid, periodID, level, 
				  "Time", datbuf);

	    for (i=0; i<size/8; i++)
	    {
		printf("%d %lf\n", i, datbuf[i]);
	    }

	    free(datbuf);
	    


	}
    }

    PTdetach(PTid);
    PTclose(ptfid);

    return;
}
